File: automatos/csharp/Automata.Tests/Automata/State.cs
namespace Automata
{
	/// <summary>
	/// Um estado de um autômato.
	/// </summary>
	public sealed class State 
	{
		/// <summary>
		/// Inicializa uma nova instância de <see cref="Automata.State"/>.
		/// </summary>
		/// <param name="name">Nome deste estado.</param>
		public State (string name, bool isFinal)
		{
			Name = name;
			IsFinal = isFinal;
		}

		/// <summary>
		/// Nome do estado.
		/// </summary>
		/// <value>O nome do estado.</value>
		public string Name { get; private set; }

		/// <summary>
		/// Determina se o estado é final ou não.
		/// </summary>
		/// <value><c>true</c> se o estado é final; caso contrário, <c>false</c>.</value>
		public bool IsFinal { get; private set; }


		public override bool Equals (object obj)
		{
			if (obj == null)
				return false;

			if (ReferenceEquals (this, obj))
				return true;

			if (obj.GetType () != typeof(State))
				return false;

			State other = (State)obj;
		
			return Name == other.Name;
		}
		

		public override int GetHashCode ()
		{
			unchecked {
				return (Name != null ? Name.GetHashCode () : 0);
			}
		}

		public override string ToString ()
		{
			return string.Format ("[State: Name={0}, IsFinal={1}]", Name, IsFinal);
		}
		
	}
}

File: automatos/java/automata/src/main/java/br/com/pucgo/automata/State.java
package br.com.pucgo.automata;

/**
 * Representa um estado de um autômato.
 *
 * Created by danfma on 22/11/14.
 */
public class State {

    private String name;
    private boolean isFinal;

    /**
     * Inicializa uma instância desta classe.
     *
     * @param name nome deste estado.
     */
    public State(String name) {
        this.name = name;
    }

    /**
     * Inicializa uma instância desta classe.
     *
     * @param name      nome deste estado.
     * @param isFinal   se o estado é final ou não.
     */
    public State(String name, boolean isFinal) {
        this(name);
        this.name = name;
        this.isFinal = isFinal;
    }

    /**
     * O nome deste estado.
     *
     * @return o nome.
     */
    public String getName() {
        return name;
    }

    /**
     * Determina se este autômato é final ou não.
     *
     * @return true, se o autômato é final; caso contrário, false.
     */
    public boolean isFinal() {
        return isFinal;
    }

    @Override
    public String toString() {
        return String.format("State: %s (final = %s)", name, isFinal ? "true" : "false");
    }

}
